pkg/protocol/http1/server.go

```diff
@@ -345,6 +385,16 @@ func (s Server) Serve(c context.Context, conn network.Conn) (err error) {
 	}
 }
 
+func NewServer() *Server {
+	return &Server{
+		eventStackPool: &sync.Pool{
+			New: func() interface{} {
+				return &eventStack{}
```

Comment:
```
New: func() interface{} {
				e := make(eventStack, 0, 4)
				return &e
			},
```

Reply:
Because of the pooling it will quickly reach the max capacity.
There is little effect to the benchmark stats, and HARDCODE number will reduced flexibility and code readability.
old:
  322678              3755 ns/op            9866 B/op         31 allocs/op
new:
  319878              3768 ns/op            9866 B/op         31 allocs/op
(The effect is not even as large as the test bias）